src/data/repositories/ConsoleLoggerRepository.ts

```diff
@@ -4,6 +4,10 @@ import { DefaultLog } from "../../domain/entities/Log";
 import { LoggerRepository } from "../../domain/repositories/LoggerRepository";
 
 export class ConsoleLoggerRepository implements LoggerRepository {
+    batchLog(logs: DefaultLog[]): FutureData<void> {
+        logs.map(log => this.log(log));
+        return Future.success(undefined);
+    }
```

Comment:
The method is async (it returns a future) so, even if the action to perform (writes to console) is not async, it has to be wrapped somehow within a Future, otherwise we are not honoring a basic Future contract: it's only executed on future.run. So the idea is:

```
        const logs$ = logs.map(log => this.log(log));
        return Future.sequential(logs$).map(() => undefined);
```

Note that log underneath has also to be refactored. Something like this:

```
        return Future.success<Error, void>(undefined).tap(() => {
            const date = new Date().toISOString();
            process.stderr.write(`[${messageType.toUpperCase()}] [${date}] ${message}\n`);
        });
```

(get Future.tap for some other EST projects)